### ISSUE TITLE
Make train_test_split interface easier 

### ISSUE THREAD

**Issue by iKintosh** [tinkoff-ai/etna#174]
### 🚀 Feature Request

Add ability to define only one parameter in TSDataset.train_test_split method

### Motivation

It is hard to define all four parameters in train_test_split method

### Proposal

At this point train_test_split(train_start, train_end, test_start, test_end) requires to define 3 parameters.

One should be able to define only one parameter like this:
---
```
train_test_split(train_end)
```
or 
```
 train_test_split(test_start)
```
then the method divides dataset into two by the value specified 

Another case user defines 2 parameters:
---
```
train_test_split(train_start, train_end)
```
train set is in range (`train_start`, `train_end`)
test set is all data after `train_end`

```
 train_test_split(train_start, test_start)
```
train set is in range (`train_start`, `test_start`)
test set is all data after `test_start`

```
train_test_split(train_end, test_start)
```
train set is all data before `train_end`
test set is all data after `test_start`

```
train_test_split(train_end, test_end)
```
train set is all data before `train_end`
test set is in range (`train_end`, `test_end`)

```
train_test_split(test_start, test_end)
```
train set is all data before `test_start`
test set is in range (`test_start`, `test_end`)

User case for 3 parameters keep as it is
---

User case for 4 parameters keep as it is
---

### Test cases

Use cases described in Proposal could be used for testing

### Checklist

- [X] Added feature request
- [X] Added motivation
- [X] Added proposal
